Galaxus.Functional/(Option)/AsyncOptionExtensions.cs

```diff
@@ -77,4 +77,144 @@ public static async Task<T> UnwrapOrElseAsync<T>(this Task<Option<T>> self, Func
 
         return option.Unwrap();
     }
+
+    /// <inheritdoc cref="Option{T}.Match" />
+    public static async Task MatchAsync<T>(this Option<T> self, Func<T, Task> onSome, Func<Task> onNone)
+    {
+        if (self.IsSome)
+        {
+            if (onSome is null)
+            {
+                throw new ArgumentNullException(nameof(onSome));
+            }
+
+            await onSome(self.Unwrap());
+        }
+        else
+        {
+            if (onNone is null)
+            {
+                throw new ArgumentNullException(nameof(onNone));
+            }
+
+            await onNone();
+        }
+    }
+
+    /// <inheritdoc cref="Option{T}.Match" />
+    public static async Task MatchAsync<T>(this Task<Option<T>> task, Func<T, Task> onSome, Func<Task> onNone)
+    {
+        var option = await task;
+        await option.MatchAsync(onSome, onNone);
+    }
+
+    /// <inheritdoc cref="Option{T}.Match" />
+    public static async Task MatchAsync<T>(this Option<T> self, Action<T> onSome, Func<Task> onNone)
+    {
+        await self.MatchAsync(async t =>
+        {
+            onSome(t);
+            await Task.CompletedTask;
```

Comment:
Why do `await Task.CompletedTask` when we could just not make the lambda async and `return Task.CompletedTask`?